### PR TITLE
CHE-1358: Normalize push conflict error message

### PR DESCRIPTION
**Pull request by vinokurig**
Old push conflict message was a little bit confused for user. Optimized code in push method.
@skabashnyuk please review

### CODE REVIEW

**Comment by skabashnyuk** [comment]
@sleshchenko ?

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java [hunk 1]
```diff
@@ -179,19 +179,15 @@
...
-                                                                          "the remote changes using pull, and then push again. \n " +
-                                                                          "Error Status: {2}. \n";
+    private static final String BRANCH_REFSPEC_SEPERATOR             = " -> ";
+    private static final String REFSPEC_COLON                        = ":";
+    private static final String KEY_COMMIT_MESSAGE                   = "Message";
+    private static final String KEY_RESULT                           = "Result";
+    private static final String KEY_REMOTENAME                       = "RemoteName";
+    private static final String KEY_LOCALNAME                        = "LocalName";
+    private static final String INFO_PUSH_ATTEMPT_IGNORED_UP_TO_DATE = "Could not push because the repository is up-to-date.";
+    private static final String ERROR_PUSH_ATTEMPT_FAILED            = "failed to push '%s' to '%s'. Try to merge " +
```
Comment (sleshchenko) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]: **F**ailed ?

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java [hunk 2]
```diff
@@ -978,60 +974,60 @@ public PullResponse pull(PullRequest request) throws GitException, UnauthorizedE
...
+                                                                 ConfigConstants.CONFIG_KEY_URL);
+        PushCommand pushCommand = getGit().push();
+        if (request.getRemote() != null) {
+            pushCommand.setRemote(remoteName);
+        }
+        List<String> refSpec = request.getRefSpec();
+        if (!refSpec.isEmpty()) {
+            List<RefSpec> refSpecInst = new ArrayList<>(refSpec.size());
+            refSpecInst.addAll(refSpec.stream().map(RefSpec::new).collect(Collectors.toList()));
+            pushCommand.setRefSpecs(refSpecInst);
```
Comment (sleshchenko) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]:
```
+            List<RefSpec> refSpecInst = new ArrayList<>(refSpec.size());
+            refSpecInst.addAll(refSpec.stream().map(RefSpec::new).collect(Collectors.toList()));
+            pushCommand.setRefSpecs(refSpecInst);
```

->

```
+            pushCommand.setRefSpecs(refSpec.stream()
+                                           .map(RefSpec::new)
+                                           .collect(Collectors.toList()));
```

?

**Comment by sleshchenko** [comment]
OK for me

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java (hunk 1 above)
Comment (vinokurig) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]: Fixed all constants

File: wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java (hunk 2 above)
Comment (vinokurig) [inline wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java]: done

**Pull request merged**